lib/jsdom/browser/js-globals.js

```diff
@@ -1,19 +1,11 @@
 "use strict";
 const vm = require("vm");
-const fs = require("fs");
-const path = require("path");
 const assert = require("assert");
 
 // Creates a list of globals from the JS environment (not the web), so that Window.js can alias them when necessary.
 // The generated list should match https://tc39.es/ecma262/#sec-global-object, to the extent V8 implements the spec.
-// We generate this at build time instead of runtime because we want to avoid the performance and memory overhead of
-// creating a new context when scripting is disabled in the JSDOM.
 
-const dest = path.resolve(__dirname, "../lib/jsdom/browser/js-globals.json");
-
-const context = vm.createContext();
-
-const globals = vm.runInContext("Object.getOwnPropertyDescriptors(this)", context);
+const globals = vm.runInNewContext("Object.getOwnPropertyDescriptors(this)");
```

Comment:
The CI logs for build-with-browser say

```
TypeError: vm.runInNewContext is not a function
```

Indeed, our [browser shim](https://github.com/jsdom/jsdom/blob/master/lib/jsdom/vm-shim.js) for the vm module does not support runInNewContext. Could you add that function there? It should be as simple as `return runInNewContext(code, createContext())`